src/components/svgSetting/svgSetting.js

```diff
@@ -1,34 +1,82 @@
-import React, { Component } from 'react';
-import '../checkBoxSelection/checkBoxSelection.scss';
-import PropTypes from 'prop-types';
+import React, { Component } from "react";
+import "../checkBoxSelection/checkBoxSelection.scss";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+import { Svgo } from "../svgSetting/svgoConfig";
+import {
+  ADD_VARIABLE,
+  UPDATE_VARIABLE,
+  DELETE_VARIABLE
+} from "../../store/actionTypes";
+
 class SvgSetting extends Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            isClicked: this.props.isSelected
+  constructor(props) {
+    super(props);
+    this.state = {
+      isClicked: this.props.isSelected,
+      arr: [],
+      svgoObject: { ...Svgo }
         };
+  }
+
+  handleSettingsClick = () => {
+    this.setState({
+      isClicked: !this.state.isClicked
+    });
+
+ 
+    let indexOf = this.props.svgOptions.findIndex(this.props.value);
+
+    if (indexOf < 0) {
+      this.props.updateStore("svgOptions",this.props.svgOptions.concat(this.props.value))
+    } else {
+      this.props.updateStore("svgOptions",this.props.svgOptions.slice(0,indexOf).concat(this.props.svgOptions.slice(indexOf + 1))
```

Comment:
so for this, for readability sake, i would say
`
let { svgOptions=[] } = this.props;
let updatedArr = indexOf < 0 
                                                    ? [...svgOptions, this.props.value ]
                                                    : [...svgOptions.slice(0, indexOf), ...svgOptions.slice(indexOf+1)];
                                                   
this.props.updateStore("svgOptions", updatedArr );
`

Reply:
Thanks, I have used spread for more readability

Reply:
`let { svgOptions=[] } = this.props;` what does this line do ? what format is this, does this also fall under spread concept.